### DISCUSSION TITLE
Unable to create radio session

### DISCUSSION THREAD

**Discussion by LLiaang** [srsran/srsRAN_Project#115] [General Help] [answered]
Hello guys, i am trying to run srsgnb files in the latest version but failed to do so due to some radio types error. The report shows that Factory for radio type uhd not found and suggests that I should select a valid type. Here is the screenshot:
![image](https://github.com/srsran/srsRAN_Project/assets/73690975/c1c6dcf3-73ea-4401-b34d-3d5ba24ddfe8)
The chinese characters :"已放弃" mean that the program was killed.

Thanks!

EDIT: I found that I didn't install UHD for ubuntu22.04 corectly by using "apt install uhd-host", but I didn't notice the report when I was building. I finally solved it by "apt install libuhd-dev".

**Comment by pgawlowicz** [comment, marked as answer]
Hi,

The example config file [gnb_rf_b210_fdd_srsUE](https://github.com/srsran/srsRAN_Project/blob/main/configs/gnb_rf_b210_fdd_srsUE.yml) is meant for usage with USRP device. 

If you want to use a virtual ZMQ-based radio emulation, please refer to this tutorial [srsRAN gNB with srsUE](https://docs.srsran.com/projects/project/en/latest/tutorials/source/srsUE/source/index.html#zeromq-based-setup). The example config files (both for gnb and srsUE) for zmq are provided.

Piotr

**Reply by LLiaang** [reply]
Thanks!
